### PR TITLE
update vaapi - vainfo

### PR DESCRIPTION
**Pull request by uknunknown**
- add enable vainfo detection checkbox in config
- defined PT_DYN_INT to load integer field from function
- PT_DYN_INT must be paired with dyn_i
- show only VAAPI codecs advertized by vainfo
- defined two invisible fields: ui and uilp used for UI enable/disable features
- check if bitrate is greater than max_bitrate (fix to avoid tvh crash)
- vp8, vp9 separate Global Quality from Quality
- load quality and max B frames filters from vainfo
- UI has several constrains or warnings implemented using vainfo
- separated 'b_depth' from 'bf'

### CODE REVIEW

**Comment by uknunknown** [comment]
This PR is a duplicate of https://github.com/tvheadend/tvheadend/pull/1530

File: src/prop.c [hunk 1]
```diff
@@ -84,6 +85,7 @@ prop_write_values
   uint32_t u32, opts;
   uint16_t u16;
   time_t tm;
+  int dyn_i;
```
Comment (Flole998) [inline src/prop.c:88]: There's no need to declare it here, the declaration can be moved to where it's actually assigned/used

Comment (uknunknown) [reply src/prop.c:88]: If you search for 'tm' (variable above) you will see was defined and used the same. I followed the same 'code style'.

Comment (uknunknown) [reply src/prop.c:88]: @Flole998 do you have any updates in this request?

Comment (uknunknown) [reply src/prop.c:88]: @Flole998 this is my last attempt :( . Let me know if you have time to work on this this request.

Comment (CvH) [reply src/prop.c:88]: @Flole998 ^^^^^^^^

**Comment by Flole998** [comment]
It looks like there is a lot of duplicate code added/in 2 files. Could you please check if that is really needed or if there is any way to reduce duplicate code? You can click on the link of the sonarcloud-bot to get more info about what it doesn't like.

**Comment by uknunknown** [comment]
I am assuming you are referring to the section vaapi.c line 348
PR1531_vaapi_c_348

We can redefine that in a single function with a parameter being passed out (codec) ... and we can add a switch for each codec ... but is really improving? In my opinion will require mode CLK to return the bits :( ... every time you call it will pass thorough the 'switch'  

- let me know if the 'duplicate code' is referring to something else.

**Comment by uknunknown** [comment]
I removed duplicate code from vaapi.c .

File: src/config.c [hunk 2]
```diff
@@ -2753,6 +2759,19 @@ const idclass_t config_class = {
       .opts   = PO_EXPERT,
       .group  = 7,
     },
+#if ENABLE_VAAPI
+    {
+      .type   = PT_BOOL,
+      .id     = "enable_vainfo",
+      .name   = N_("Enable vainfo detection"),
+      .desc   = N_("Enable vainfo detection in order to show only "
+                   "encoders that are advertized by VAAPI driver.\n"
```
Comment (Flole998) [inline src/config.c:2768]: advertised, not advertized I think

**Pull request closed without merging**